src/ai/core/signal_processor.py

```
@@ -0,0 +1,172 @@
+                    competition_intensity,
+                    market_data
+                )
+            }
+        except Exception as e:
+            print(f"Error forecasting competition: {str(e)}")
+            return {}
+
+    def calculate_market_pressure(
+        self,
+        competition_intensity: float,
+        market_data: Dict
```

Ruff (0.8.2)

Use `dict` instead of `Dict` for type annotation

Replace with `dict`

(UP006)